compiler/rustc_target/src/spec/apple_base.rs

```diff
@@ -284,6 +284,22 @@ fn ios_lld_platform_version() -> String {
     format!("{major}.{minor}")
 }
 
+pub fn mac_catalyst_deployment_target() -> &'static str {
+    // Consider making this user-changeable in the future.
+    "14.0"
+}
+pub fn mac_catalyst_sdk_version() -> &'static str {
+    // Consider making this user-changeable in the future
+    "16.2"
+}
+pub fn mac_catalyst_llvm_target(arch: &str) -> &'static str {
```

Comment:
Could we use a boolean or enum instead of `str`? 

There're only two cases for Mac Catalyst